pkg/nsx/services/subnet/subnet.go

```diff
@@ -166,12 +167,12 @@ func (service *SubnetService) CreateOrUpdateSubnet(obj client.Object, vpcInfo co
 				updatedSubnet.SubnetDhcpConfig = nsxSubnet.SubnetDhcpConfig
 				// Only update gateway_addresses, dhcp_server_address, and connectivity_state from AdvancedConfig
 				if nsxSubnet.AdvancedConfig != nil {
-					if updatedSubnet.AdvancedConfig == nil {
-						updatedSubnet.AdvancedConfig = &model.SubnetAdvancedConfig{}
+					updatedSubnet.AdvancedConfig = &model.SubnetAdvancedConfig{
+						GatewayAddresses:    nsxSubnet.AdvancedConfig.GatewayAddresses,
+						DhcpServerAddresses: nsxSubnet.AdvancedConfig.DhcpServerAddresses,
+						ConnectivityState:   nsxSubnet.AdvancedConfig.ConnectivityState,
+						StaticIpAllocation:  nsxSubnet.AdvancedConfig.StaticIpAllocation,
```

Comment:
I feel this should be fine for now as EnableVlanExtension shall always be false for auto-created Subnet. But we may be careful when new field is introduced to SubnetAdvancedConfig that we may overwrite some value by nil when creating this.